### ISSUE TITLE
I maybe made you raise the xp cost of the entire game because I messed up my math

### ISSUE THREAD

**Issue by Zironic** [Arx-Game/arxcode#50]
So this is the original XP tax:
500*(1500/500)+50*(2000/500)+30*(2050/500) =1823
30*(1500/500)+50*(1530/500)+500*(1580/500) =1823

And this is the current XP tax:
500*(2000/500)+50*(2050/500)+30*(2080/500) =2329.8
30*(1530/500)+50*(1580/500)+500*(2080/500) =2329.8

As you can see, it was order agnostic to start with and now I wonder why I was so sure it wasn't.

**Comment by TehomCD** [comment]
I kinda meant to adjust the curve a little bit in xp overall due to inflation from first impressions and all, and if people had the _impression_ that buying legendaries earlier was better due xp tax, then making it pretty explicitly clear that wasn't the case isn't the worst thing in the world.

Right now I'm working on stuff to let staff add a catchup mechanic for additional skill points in character creation. So what I might do is put that in then let the new count-the-xp-spend-in-tax thing remain just because it might be more intuitive for people, maybe.

**Comment by rozzingit** [comment]
Whether or not buying legendaries early makes any difference in your tax, people will still do it early because it means the least amount of time basically sitting and adding NO SKILLS to your sheet. Your rate of skill addition afterwards might be slower, but it's a lot easier to feel progress. Versus that static feeling of saving up XP forever for that 6.

**Comment by TehomCD** [comment]
Maybe so, though I wonder how universal that feeling would be - my experience is that a relatively small proportion of characters have bought legendaries. I had more than one person complain that they felt they had cheated themselves by not buying a legendary early, so even if it's a matter of perception, I don't think it's a terrible thing to address.

**Comment by Zironic** [comment]
Well, when legendaries were not properly accounted for by the tax code, then it was always optimal to buy the legendary first because that way you were basically never taxed at all compared to the multiple thousands of tax trying to buy a legendary later amounts to. The legendary tax burden got somewhat lowered now too, since with old costs and 50 dots invested you'd be paying 1000*(1000/500) = 2000 while now you pay 500*(1500/500) = 1500.
 
I think it's worthwhile to consider how common you want legendaries to be. Outside of training purposes, I feel like they're not very strong mechanically the way Arx is designed. They feel like a cool thing that allows characters to stand out and demonstrate a touch of mechanical uniqueness, they certainly don't provide you with 50 dots worth of power (That's enough to buy three 5's, one 2 and two 1's), still I really want my character to have a 6 in their main skill because it would make them basically the coolest at it.

**Comment by rozzingit** [comment]
> Well, when legendaries were not properly accounted for by the tax code, then it was always optimal to buy the legendary first because that way you were basically never taxed at all compared to the multiple thousands of tax trying to buy a legendary later amounts to.

Right, but that information wasn't _actually_ that commonly known. I saw a lot of people assume that the tax would just be crazy nuts for suddenly dropping 1000+ xp onto your sheet, whereas in reality it wasn't a proportional increase in your tax. It's the few people who took the plunge to buy the skill that found it out, and I'm sure that not all of them knew about the tax difference going in. It's starting to get around _now_, but I definitely think people just figured before that "well, it'll be more expensive if I buy other skills first," so they bought the skills most necessary and then buckled down to save for their 6. There are always people who crunch the numbers, but there are also plenty of people who really _don't_. And maybe the tax would have come out the same, but the feeling of saving up 1000 xp versus 2000 xp is very different mentally.

**Comment by Zironic** [comment]
Also, I knew I wasn't completely insane with my tax plan proposal. Here we go.

My tax is dot agnostic, that is if you buy X amount of dots, you always spend Y amount of XP.
My plan:
500 * (750/500)+10 * (1005/500)+40 * (1030/500)=852.5
30 * (515/500)+20 * (540/500)+500 * (800/500)=852.5

500 * (750/500)=750
150 * (575/500)+150 * (725/500)+150 * (875/500)+50 * (975/500)=750


Current XP plan:
500 * (1000/500)+10 * (1010/500)+40 * (1050/500)=1104.2
30 * (530/500)+20 * (550/500)+500 * (1050/500)=1103.8

500 * (1000/500)=1000
150 * (650/500)+150 * (800/500)+150 * (950/500)+50 * (1000/500)=820

What that means is that if you calculate the Tax my way (You divide the cost of the skill you're buying by 2, then add it to tax), buying a Legendary always costs the same as three 5's and 5 1's. While the current way it's kind of hard to calculate. It also means everyone who has spent X amount of (pre-training) XP are at the exact same tax.

**Comment by TehomCD** [comment]
As far as intent goes, my understanding is that legendaries are supposed to be enormously inefficient to consider buying from any mechanical standpoint - it's supposed to be something you pursue only if you want to distinguish your character as incredibly specialized and skilled in one specific area and well known for that.

**Comment by Zironic** [comment]
In terms of making things better for newer players, these changes have some unfortunate grandfather implications, I'll demonstate with some math and three hypothetical characters. First we have Charlie, he's a veteran player who didn't buy a legendary. Then we have Bob, he heard the rumor that you should buy legendaries first and asked for a full respec so he could do so. Then we have Alice, she aspires to one day be like Bob.

First to make this math a lot more readable and easier to type, we will establish the tax accounted for price to raise a skill to 5. To do this we simply do

Old Tax:
10*(500/500)+20*(510/500)+30*(530/500)+40*(560/500)+50*(600/500)= X*(500/500) = 167
X = 167
10*(1850/500)+20*(1860/500)+30*(1880/500)+40*(1910/500)+50*(1950/500)= X*(1850/500) = 572
x = 154

New Tax:
10*(510/500)+20*(530/500)+30*(560/500)+40*(600/500)+50*(650/500)= 178
Since the new tax is effectively a flat cost raise, it means buying 0->5 is always going to cost 11xp extra then it used to (before rounding).

Combat skills are 4x worse off, at 44xp extra.
20*(500/500)+40*(520/500)+60*(560/500)+80*(620/500)+100*(700/500)=368
20*(520/500)+40*(560/500)+60*(620/500)+80*(700/500)+100*(800/500)=412

With that known, we can establish that in the old tax system, buying 10 dots to 5 would cost you approximatly

sum (168-1.3j)*((350+j150)/500) from j=1 to 10  = 3732
166.7 + 215.02 + 262.56 + 309.32 + 355.3 + 400.5 + 444.92 + 488.56 + 531 + 573.5

And with the new tax system it would cost you 110 xp more.
 
That means that if Alice wants to earn the sheet of Charlie, she has to earn a very similar amount of total XP as Charlie which is good. However with Bob since he only spent 1000xp and 60 tax for his Legendary, Charlie is actually behind him in XP by 1000*(1700/500)-1000*(560/500) = 2280 which means Bob actually had 63% more XP on his sheet then Charlie even though they've earned the same amount of XP in play. The legendary effectively means he could only raise 8 dots to 5 instead of 10.
 
And if Alice wanted to get the same sheet as Bob, that is 8 5's and one legendary. She would have to pay about 2650+500*(2100/500)=4750, which is 1000 more XP then Bob paid for that sheet. If Bob has a combat legendary, it's 2650+1000*(2600/500)=7850 which is 4100 xp more then Bob paid for it.

**Comment by TehomCD** [comment]
There's few enough people with legendaries where it wouldn't be that arduous to refund them all. I don't have much of an opinion on that personally - whether it'd feel more unfair to remove things people already had, or whether it'd feel worse for people who feel cheated due to someone having them already at a reduced rate.

I'd kind of lean towards sympathizing more with the latter, though there's cases of people who did significant amounts of RP around getting a legendary, including `@actions` about it, and I wouldn't want to invalidate stories.

**Comment by rozzingit** [comment]
Considering the changes, I'd _think_ that most people who bought a non-combat legendary would be able to buy it back in short order if they were refunded the full cost they paid. I guess possibly not if they've built their sheet a lot since then in a way that would inflate their tax higher. On the other hand, the few people who bought combat legendaries might indeed be stuck unable to buy back their legendary. Which I do think would be pretty demoralizing for people who spend the time saving.

**Comment by Zironic** [comment]
Honestly i think that the way that XP tax works, the combat legendary has no need to be twice the base price of the non-combat legendary if it's intended to be roughly twice as expensive. In fact if you want it to be 2x as expensive when tax is accounted for, it should cost 900xp. If you want it to be 1.5x as expensive when tax is accounted for, it should cost 700

Cost by the time a character has bought 200 dots( worth of skills after chargen.
500 * (3000/500)=3000
700 * (3200/500)=4480
900 * (3400/500)=6120
1000 * (3500/500)=7000

I think a strong argument could be made for very high flat-cost entirely untaxed legendaries because that would make it so they would only really be viable to buy towards the end of your characters lifespan after which they've presumably already made their niche through RP and you'd entirely remove that feeling people are complaining about where they feel they get further away from their legendary the more they improve their character in other ways (It would still be true, but the price of the legendary wouldn't go up in front of their eyes)

**Comment by Sritterbush** [comment]
Bit late to the conversation, but a though on the breadth of impact on these changes. Tehom mentioned there are few people with legendaries already, I was wondering if there is a good metric to estimate how many people were working towards legendaries. Perhaps a tally of how many people had 500+ xp unspent, which would be a notable chunk to be holding onto. 

That may not ultimately change how this is handled, but given the time involved in achieving a legendary, it seems as though the number of people who were working towards that goal might be estimated just for some sense of impact on the playerbase.

**Comment by Sritterbush** [comment]
To add to Rozzingit's point about demoralizing those who already had their combat legendaries, I believe it is demoralizing for those who were working towards a combat legendary to see that it is now unobtainable without a far longer road, but that others were allowed to keep legendaries who already obtained them before the change. It builds a sense of haves and have-nots, which doesn't seem to be the goal here. If the worry is for story being harmed, then perhaps stories should be disconnected from xp investment, as was also proposed.

**Comment by Precisi** [comment]
A thought -- Maybe add some kind of 'bank' for XP.. only in that if you're saving up for a big buy you could deposit into it, maybe like layaway! Personally, the reason I haven't saved for a legendary is that I always get bored and end up buying a skill for fun, but if I had a conscious bucket to dump my xp into I could look at my progress and go 'yay, I'm a little closer'. 

This would seem like a superfluous feature, except that a) it would give players a week to week sense of progress towards an XP goal, and b) it would give staff a way to SEE what people are working/interested on. 

Just an idea.

**Comment by earthsend** [comment]
@Precisi  I like that suggestion, because it adds a sense of progression to long-term XP goals.

**Comment by TehomCD** [comment]
What I might do is when skill raises take training time, rather than spending xp all at once you can put it in as-needed to increase the maximum progress it can achieve. So you could put in 10% of the required xp for a legendary and progress would stop at 10% of the time required until you spend more.

**Comment by Precisi** [comment]
> 
> 
> What I might do is when skill raises take training time, rather than spending xp all at once you can put it in as-needed to increase the maximum progress it can achieve. So you could put in 10% of the required xp for a legendary and progress would stop at 10% of the time required until you spend more.

I dig it, but what skills require time?

**Comment by TehomCD** [comment]
None currently, all eventually. Training will reduce time required, not xp.

**Comment by earthsend** [comment]
Will that replace XP tax?

**Comment by TehomCD** [comment]
No, though I'd probably restructure how tax is calculated and make it apply to time also.

**Comment by earthsend** [comment]
It might get a little finnicky, if XP and time are both used as purchasing barriers. If you depart from the 'time is money' perspective, XP is a way to measure time commitment to the goal of advancing a skill. You'd be charging double tax, effectively: both for the time delay, and the cost multiplier. This might be moot if you're reducing how XP tax is calculated to account for time, though: making immediate purchases more expensive in the long-run than a gradual buy-in. 

Might I also recommend expanding the buffer to three skills instead of just one? And maybe a more permanent way to record your teachers.

**Comment by Sritterbush** [comment]
Perhaps have skill raises cost time and no XP, but XP reduces the time up to some cap implying full XP investment?

**Comment by Zironic** [comment]
Trading XP for time sounds like the kind of thing that would generate frustration for a lot of people, I don't like creating situations where a player can feel like 'If I was only more patient, my character would be better off'.

 In other games which have had training time it's been somewhat common to allow multiple things to be trained at once, usually from different categories like one skill, one stat and one ability, it's really just a question of tuning the training times to a point where they feel satisfying. If someone really wants to train faster, even faster then they can with a trainer, they could probably do an @action to learn it through magic or plot.

**Comment by earthsend** [comment]
XP and time are equivalent currencies. You get XP on a weekly basis, and the higher you want to go on a skill, the more time you'll spend saving up. If you're adding a delay to skill acquisition, you might as well reward the player for being patient by virtue of lowered costs. You buy into the initial investment, wait your time, and get the skill for cheaper. Or you can not bid your time at all and spend a little more to get it right away. XP is infinite, after all; its value just depreciates as the character acquires more points.  Time, on the other hand, is what experience points try to abstract.

**Comment by TehomCD** [comment]
Making them not directly tied to one another avoids some problems with their role as incentives, imo. It also provides far more flexibility in terms of soft-capping character growth without being too draconian. You could easily say "You can only earn X amount in Y time" but bottlenecking spending is far gentler.

**Comment by earthsend** [comment]
Won't that change the aspect of making the game's systems equitable entirely, though? There will be a number of characters who could spend XP right away, at any time, and then there will be others who will not have been able to spend XP right away and will be much worse in comparison to the  first group. They won't even be able to catch up. I could understand changing this paradigm with a reset, but that leads to another balancing act.

Which isn't to say it's a 'group vs group' situation, but hypothetically speaking, there's about four ways this change could go, in the long term. I'm just pointing out the commonality of the worst two.  Maybe a compromise is to calculate a certain amount of Earned Lifetime XP as 'immediate spends OK' and start capping from that point onwards.

**Comment by TehomCD** [comment]
Yeah, there'd be a reset accompanying this. If training time is proportional to tax, then 'catchup' xp would function the same way, with spends early being instant or near-instant, and then becoming slower as people get over the hump, and then stretching into eternity for the dinosaurs. One of the benefits for doing something like this is you could then make reducing training time into a commodity that people could buy in other ways in addition to training, such as AP spends, retainer assistants, or other incentives. Having multiple bottlenecks allow you to create sinks for different forms of game currencies/resources more safely due to another cap existing.

**Comment by earthsend** [comment]
Yeah, that sounds reasonable to me - it allows people just enough room to catch up, and then when they're reasonably on the same level as the others, it slows down.

**Comment by dreamsofgold** [comment]
I originally sent this as a +request in-game, but Apostate asked me to post it to the github for posterity's sake for the postgres cutover, so here it is:

"So, this is likely not something you want to deal with right now, but in light of the work you guys are gonna have to do for the postgres cutover, I figured this is as good an opportunity as any.



As you may have noticed, both through my goals and my rfrs, I spun Amund's entire story around training and becoming a blademaster of note. Part of this is attaining a combat legendary, medium wpn 6, and then starting to work on teaching and capping a few skills at five in order to achieve the overarching goal.



The problem is, I'm running up a very, very steep hill. And that hill is the double-cost for combat legendaries. See, with the base XP changes, plus the refactor of the multiplier for combat legendaries, instead of the nominal XP cost of 1000 as is written in the code, it costs me three thousand XP to get there. I am currently sitting on a pile of 1800 XP.



It's been said before that the order of spends don't matter, and that legendaries need to be end-game... but they do. Especially now that legendaries add to XP tax, which means that you can't really just do them last if you want. You have to do them first, and rely on either gear or your lucky stars to survive long enough for three thousand XP.



And the problem is that anyone who can only dedicate maybe 10 hours a week or less to the game cannot attain that goal in less than 3 years. Let's assume a person earns 20 XP a week through any combination. You still need 150 weeks in order to achieve your goal. 150 weeks split by four equals 37 and a half months.



That is roughly three years, one month, fifteen days to achieve the legendary. Understand, then, that that is a lot of time to dedicate to a single purchase in the game, and while I understand the design in making legendaries harder to attain, this has made them inconvenient unless you can somehow pull 40 or 60 XP a week. As a frame of reference, the top 20 earners only get around 25 to a maximum of 35 a week, tops.



It isn't a deal breaker, and it can be quite an exercise in cunning to use only a character's starter sheet to overcome challenges, but there comes a time where even that hits a plateau. It is hard to see other people's characters progressing and seeing yours encased in ice until that fateful day it dies or it vaults over the peak and finally makes the goal.



So, as a tl;dr, what I'm asking is to reconsider the refactoring of combat legendaries to something like 1.25 or 1.30 instead of 2.00 because this would make the goal much less of a sacrifice on the player."
